### ISSUE TITLE
add season_sequence_number to a episode output

### ISSUE THREAD

**Issue by SatoshiMiamoto** [crunchy-labs/crunchy-cli#430]
When I download anime episodes, some series have very incorrect numbering. Looking at the episode requests on the site

https://www.crunchyroll.com/content/v2/cms/seasons/GY3VWX3MR/episodes

I noticed that there is a property called `season_sequence_number `that would solve this problem. Is it possible to implement this in the episode outputs?

thanks 
![image](https://github.com/crunchy-labs/crunchy-cli/assets/69407195/672b3604-9b73-4d0d-8f0f-b7ad85c55352)

**Comment by bytedream** [comment]
This isn't possible, sadly `season_sequence_number` does not like you'd expect it, it has nothing to do with the actual season number.

E.g. season 2 of Ancient Magus Bride:
![image](https://github.com/crunchy-labs/crunchy-cli/assets/63594396/e21f92ef-66bc-4bce-8873-68c98570b111)

**Comment by SatoshiMiamoto** [comment]
Is there a way to catch the season? in the case of Demon Slayer gets too discrepant from reality. Season 1 turns into 53

**Comment by SatoshiMiamoto** [comment]
From brazil pt-BR i get the url from the webiste

**Comment by Elwador** [comment]
i started using the identifier to get the season numbers - at least for now i haven't had any issues with that

**Comment by SatoshiMiamoto** [comment]
> i started using the identifier to get the season numbers - at least for now i haven't had any issues with that

the season_number identifier?

**Comment by Elwador** [comment]
Every season has an `"identifier": "GY5P48XEY|S4",`. For example, for the Demon Slayer seasons:

- `GY5P48XEY|S1`
- `GY5P48XEY|S2C1`
- `GY5P48XEY|S2C2`
- `GY5P48XEY|S3`
- `GY5P48XEY|S4`

This way, it is relatively easy to get season numbers that are not strange, like 53, even though there are only 5 seasons. It also helps with the One Piece season numbers since I had a lot of duplicates there as well.

**Comment by bytedream** [comment]
Oh yea I totally forgot about this. When the `--experimental-fixes` flag is set, the season number from the identifier is used, at least if it can be parsed to a number. Unfortunately the last time when I touched it was 1+ year ago, so it does not work when the identifier is e.g. `S2C1`, `M2` (so basically it can only parse if the season in the identifier is `S<number>`)

**Comment by Elwador** [comment]
Yeah, I think `M` is for movies, and then there is also `OVA` for some. Maybe there are even more variants; who knows, it is Crunchyroll 😅

**Comment by SatoshiMiamoto** [comment]
I went to run a test without the flag and it returned correctly; for some reason, it seems they fixed the references internally. Via CLI it is now correct, but looking at the request directly from the site, it's still showing the value 53. I'm not exactly sure what happened, but I will start using the flag --experimental-fixes.

![image](https://github.com/crunchy-labs/crunchy-cli/assets/69407195/cb045036-29fc-4bd9-a1f0-4d30ff7d5a08)
![image](https://github.com/crunchy-labs/crunchy-cli/assets/69407195/69093e3b-ff36-4ce9-9685-a5a4cb990eb8)

**Comment by SatoshiMiamoto** [comment]
I discovered the problem, but I don't know the cause. When I use the flag --audio ja-JP in the search, it returns the values with the incorrect season.

`./crunchy-cli search  --output "{{series.title}} {{season.number}} {{episode.number}} {{episode.sequence_number}} {{episode.id}} {{episode.title}}"  https://www.crunchyroll.com/series/GY5P48XEY/demon-slayer-kimetsu-no-yaiba --audio ja-JP`

![image](https://github.com/crunchy-labs/crunchy-cli/assets/69407195/a846586e-1c2c-4c81-acdc-b126acc43502)

However, when I don't specify the audio flag, it shows the seasons correctly.

```
 ./crunchy-cli search  --output "{{series.title}} {{season.number}} {{episode.number}} {{episode.sequence_number}} {{episode.id}} {{episode
.title}}"  https://www.crunchyroll.com/series/GY5P48XEY/demon-slayer-kimetsu-no-yaiba
```
![image](https://github.com/crunchy-labs/crunchy-cli/assets/69407195/9e5f0f0c-45d6-4371-bf59-15f31081f3a3)
